libs/frontend/feature-render-builder/src/lib/render-builder.ts

```diff
@@ -273,7 +276,7 @@ export const computePropsContent = (
     ...data.attrs,
     ...builderStyleProps?.builderProps,
     ...events.native,
-    class: data.mixins.concat(builderStyleProps?.builderClass ?? []),
+    class: data.mixins.concat(component.id, builderStyleProps?.builderClass ?? []),
```

Comment:
Hi @sampullman 

Sorry for the delay! Currently we use `#test-c-0` to apply component custom styles, and use `.reusable-style-0` to apply reusable styles. Since the `#id` selector has a higher weight than the `.class` selector in CSS, and we actually flattened those styles before using it to generate the `<style>` tag, component custom styles will always override reusables styles not only in the builder, but also in preview and live site.

So I changed selector from `#id:hover` to `.id._hover` (class-based pseudo class) in build mode so that component custom styles and reusable styles always have the same weight. If they have the same weight, component custom styles will always override reusable styles because `<ReusableStyle />` is placed/rendered before `ComponentStyle` in all 3 modes (build, preview, and live).

For preview & live sites, the only change being made was that `#id` now becomes `.id` due to the weight issue mentioned above. But `:hover` is still `:hover`; it is not converted to `.hover` in preview & live site. The class-based pseudo class is only visible in the builder because we always want to see the corresponding styles when active pseudo-class changes.

I think I didn't explain it very well, sorry about that! 😵. Feel free to let me know if you need further explanation!

Reply:
It makes sense to me, thanks for the effort it took to solve this one!

When I was last working on style rendering I think I rushed too much and didn't take time to think through all the precedences. I probably should have drawn a diagram first!